### PR TITLE
Error with better messages

### LINKED ISSUE #21967: Better Error Messages (several scripts)
## Reason
We see some common errors on Discussions that we might not, if we printed useful error messages. While this PR isn't going to solve them all, two I want to fix:

Peacock specifying the loading of some relic module from our redistributable moose packages.
The TestHarness spitting out a stack trace when `./app_name --json` fails.

## Design
Produce useful messages.

## Impact
Possibly less discussion posts

### PR DESCRIPTION
**Pull request by milljm**
Return proper error when application can't report registered objects.
Return an actual solution when running Peacock without proper dependencies.
Add unittest to test for proper json dumps.

Closes #21967

### CODE REVIEW

File: python/TestHarness/tests/test_FailedJSON.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+        """
+        out = self.mocked_output('**START JSON DATA**\n{}**END JSON DATA**\n', False)
+        self.assertRegex(out, r'.*?AnalyticalIndicator not found in executable')
+
+    def testBadJSONOutput(self):
+        """
+        Test for bad json output
+        """
+        out = self.mocked_output('**START JSON DATA**\n{badjson}**END JSON DATA**\n', True)
+        self.assertRegex(out, r'.*?produced invalide JSON output')
```
Comment (cticenhour) [inline python/TestHarness/tests/test_FailedJSON.py:44]:
```suggestion
        self.assertRegex(out, r'.*?produced invalid JSON output')
```

File: python/TestHarness/util.py [hunk 2]
```diff
@@ -797,9 +798,17 @@ def getExeJSON(exe):
...
-    return json.loads(output)
+    try:
+        output = output.split('**START JSON DATA**\n')[1]
+        output = output.split('**END JSON DATA**\n')[0]
+        results = json.loads(output)
+    except IndexError:
+        print(f'{exe} --json, produced an error during execution')
+        sys.exit(1)
+    except json.decoder.JSONDecodeError:
+        print(f'{exe} --json, produced invalide JSON output')
```
Comment (cticenhour) [inline python/TestHarness/util.py:809]:
```suggestion
        print(f'{exe} --json, produced invalid JSON output')
```

File: python/peacock/CheckRequirements.py [hunk 3]
```diff
@@ -7,10 +7,15 @@
...
 
+from platform import python_version
+
 def print_error(msg):
-    print("\nError starting peacock: %s\n" % msg)
-    print("You need to use the miniconda module. Run:")
-    print("\tmodule load miniconda")
+    python_short = '.'.join(python_version().split('.')[:2])
+    print(f"\nError starting peacock: {msg}"
+           "\n\nYou may need to either create or load an environment providing PyQt, VTK, etc"
```
Comment (cticenhour) [inline python/peacock/CheckRequirements.py:15]:
```suggestion
           "\n\nYou may need to either create or load an environment providing PyQt, VTK, etc."
```

File: python/peacock/CheckRequirements.py [hunk 4]
```diff
@@ -7,10 +7,15 @@
...
+           "\nWe provide a conda package with the required dependencies:"
```
Comment (cticenhour) [inline python/peacock/CheckRequirements.py:16]:
```suggestion
           "\nThe MOOSE development team provides a conda package with the required dependencies:"
```

Prefer using this over "we", but that's personal opinion. Feel free to disregard this one.

**Review by cticenhour — CHANGES_REQUESTED** [review]
I am unaware of the general context of this PR, but did notice a few typos when scanning the diff. I am happy to give a more thorough review, but someone else may want to jump in who knows the need for this change better.

File: python/peacock/CheckRequirements.py (hunk 4 above)
Comment (milljm) [reply python/peacock/CheckRequirements.py]: Na, I like the suggestion over 'we' as well

**Comment by milljm** [comment]
> I am unaware of the general context of this PR, but did notice a few typos when scanning the diff. I am happy to give a more thorough review, but someone else may want to jump in who knows the need for this change better.

Its mostly to help elevate things like: 
https://github.com/idaholab/moose/discussions/20459#discussion-3909816
https://github.com/idaholab/moose/discussions/21962#discussion-4345451

**Review by cticenhour — APPROVED** [review]

**Pull request merged**